python/lsst/daf/io/table/sql.py

```diff
@@ -0,0 +1,938 @@
+    read_sql_query : Read SQL query into a afw table
+
+    """
+    sql_io = sql_io_builder(con)
+
+    try:
+        _is_table_name = sql_io.has_table(sql)
+    except:
+        _is_table_name = False
+
+    if _is_table_name:
+        sql_io.meta.reflect(only=[sql])
+        return sql_io.read_table(
+            sql, column_dtypes=column_dtypes, coerce_float=coerce_float,
+            parse_dates=parse_dates, columns=columns, chunksize=chunksize)
+    else:
+        return sql_io.read_sql(
+            sql, params=params, column_dtypes=column_dtypes,
+            coerce_float=coerce_float, parse_dates=parse_dates,
+            chunksize=chunksize)
+
+
+def to_sql(catalog, name, con, schema=None, if_exists='fail',
+           chunksize=None, dtype=None):
+    """
+    Write records stored in a afw table to a SQL database.
+
+    Parameters
+    ----------
+    catalog : afw table
+    name : string
+        Name of SQL table
+    con : SQLAlchemy connectable(engine/connection) or database string
+        URI. Using SQLAlchemy makes it possible to use any DB supported
+        by that library.
+    schema : string, default None
+        Name of SQL schema in database to write to (if database flavor
+        supports this). If None, use default schema (default).
+    if_exists : {'fail', 'replace', 'append'}, default 'fail'
+        - fail: If table exists, do nothing.
+        - replace: If table exists, drop it, recreate it, and insert data.
+        - append: If table exists, insert data. Create if does not exist.
+    chunksize : int, default None
+        If not None, then rows will be written in batches of this size at a
+        time.  If None, all rows will be written at once.
+    dtype : single SQLtype or dict of column name to SQL type, default None
+        Optional specifying the datatype for columns. The SQL type should
+        be a SQLAlchemy type.
+        If all columns are of the same type, one single value can be used.
+
+    """
+    if if_exists not in ('fail', 'replace', 'append'):
+        raise ValueError("'{0}' is not valid for if_exists".format(if_exists))
+
+    sql_io = sql_io_builder(con, schema=schema)
+    sql_io.to_sql(catalog, name, if_exists=if_exists, schema=schema,
+                  chunksize=chunksize, dtype=dtype)
+
+
+def has_table(table_name, con, schema=None):
+    """
+    Check if the Database has named table.
+
+    Parameters
+    ----------
+    table_name: string
+        Name of SQL table
+    con: SQLAlchemy connectable (engine/connection).
+        Using SQLAlchemy makes it possible to use any DB supported by that
+        library.
+    schema : string, default None
+        Name of SQL schema in database to write to (if database flavor supports
+        this). If None, use default schema (default).
+
+    Returns
+    -------
+    boolean
+    """
+    sql_io = sql_io_builder(con, schema=schema)
+    return sql_io.has_table(table_name)
+
+
+table_exists = has_table
```

Comment:
did you leave this on purpose? some comment about aliasing the function name would be appropriate, if that's what you're indenting to do here.